padinfo/padinfo.py

```diff
@@ -1537,13 +1546,13 @@ async def jpyoutube(self, ctx, *, search_text):
         if not sds:
             return await ctx.send(f"No dungeons found. This command uses `/` or `,` as an optional delimiter to specify a leader, "
                                   f"maybe try again?")
-
+        
         dungeons = self.make_dungeon_dict(sds)
 
         menu = ClosableEmbedMenu.menu()
-        props = JpYtDgLeadProps(sorted(dungeons.values(), key=lambda d: d['idx']), monster)
+        props = DgLeadProps(sorted(dungeons.values(), key=lambda d: d['idx']), monster)
```

Comment:
How come this line is changed? And where is it coming from? I'm not following here

Reply:
This line doesn't need to be changed technically since the JpTwtDgLeadProps are the same as the JpYtDgLeadProps (JpTwtDgLeadProps is subclass of JpYtDgLeadProps). I have them different in case they do need to be different one day, so in this function that both `^jpyoutube` and `^jptwitter` will call, a generic DgLeadProps is used, but I can change it back to what it was before!

Reply:
If this is the case, then I think the correct thing to do is to parametrize the classes in the function call and have each caller send their own particular values here. Because either:

- they're the same always in which case this is sort of a TODO for when they're different and then don't change it yet
- they're different in which case this is a bug because each separate caller *requires* its own different props & view